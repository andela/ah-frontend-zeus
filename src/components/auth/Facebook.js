import React from 'react';
import FacebookLogin from 'react-facebook-login';
import { connect } from 'react-redux';
import { responseFacebook } from '../../actions/socialAction';

export class Facebook extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <FacebookLogin
        appId="328736491284498"
        autoLoad={false}
        fields="name,email,picture"
        callback={this.props.responseFacebook}
        size="small"
        icon="fa-facebook"
        textButton="FACEBOOK"
      />
    );
  }
}

export default connect(
  null,
  { responseFacebook }
)(Facebook);
