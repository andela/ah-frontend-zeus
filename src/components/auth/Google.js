import React from 'react';
import GoogleLogin from 'react-google-login';
import { connect } from 'react-redux';
import { responseGoogle } from '../../actions/socialAction';


export class Google extends React.Component {
  constructor() {
    super();
  };

  render() {
    return (
      <GoogleLogin
        buttonText="GOOGLE"
        prompt="consent"
        redirectUri="/"
        autoLoad={false}
        responseType="id_token"
        clientId="691552019879-rso0i7gaschrtmtv5d2rci49elj82eht.apps.googleusercontent.com"
        onSuccess={this.props.responseGoogle}
        onFailure={this.props.responseGoogle}
        size="metro"
        icon={true}
      />
    );
  }
}
export default connect(null,{ responseGoogle })(Google);
