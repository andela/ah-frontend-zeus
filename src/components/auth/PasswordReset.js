import React, { Component } from 'react';
import classnames from 'classnames';
import { passwordReset } from '../../actions/authPasswordReset';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

export class PasswordReset extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  onSubmit(event) {
    event.preventDefault();
    const userEmail = {
      email: this.state.email
    };
    this.props.passwordReset(userEmail);
  }

  render() {
    const { errors } = this.state;

    return (
      <section id="passwordreset" className="flex-grow-1">
        <div className="container">
          <div className="row">
            <div className="col-md-8 m-auto">
              <h3 className="display-4 text-center my-4">
                Request a Password reset link
              </h3>
              <form noValidate onSubmit={this.onSubmit}>
                <div className="form-group">
                  <input
                    type="email"
                    className={classnames(
                      'form-control form-control-lg',
                      {
                        'is-invalid':
                          errors.message &&
                          errors.message !==
                            'Check your email for the password reset link'
                      },
                      {
                        'is-valid':
                          errors.message &&
                          errors.message ===
                            'Check your email for the password reset link'
                      }
                    )}
                    placeholder="Enter Your Email Address"
                    name="email"
                    value={this.email}
                    onChange={this.onChange}
                  />
                </div>
                {errors.message !==
                  'Check your email for the password reset link' && (
                  <div className="invalid-feedback">{errors.message}</div>
                )}
                {errors.message ===
                  'Check your email for the password reset link' && (
                  <div className="valid-feedback">{errors.message}</div>
                )}
                <input
                  type="submit"
                  className="btn btn-primary btn-block mt-4 zeus-color"
                />
              </form>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

PasswordReset.propType = {
  passwordReset: PropTypes.func.isRequired,
  errors: PropTypes.object
};

const mapStateToProps = state => ({
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { passwordReset }
)(withRouter(PasswordReset));
