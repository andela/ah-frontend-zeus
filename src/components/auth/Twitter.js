import React from 'react';
import TwitterLogin from 'react-twitter-auth';
import { push_uniq } from 'terser';

class Twitter extends React.Component {
  onSuccess(response) {
    console.log(response)
    const token = {
      auth_token: response.accessToken
    };

    fetch('https://zeus-staging.herokuapp.com/api/social/auth/twitter/', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      CORS: 'no-cors',
      body: JSON.stringify({ user: token })
    })
      .then(res => res.json())
      .then(data => {
        if (data){
          window.location = ('/')
        }
      })
      .catch(error => console.log(error));
  }

  render() {
    return (
      <TwitterLogin
        appId='gvoXkTkL8BZzddFeUrkJuldpL' 
        loginUrl="https://api.twitter.com/oauth/authorize"
        onFailure={this.onFailed}
        onSuccess={this.onSuccess}
        requestTokenUrl="https://zeus-staging.herokuapp.com/api/social/auth/twitter/"
        showIcon={true}
        forceLogin={true}
      />
    );
  }
}

export default Twitter;
