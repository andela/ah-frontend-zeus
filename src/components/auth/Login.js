import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { loginUser } from '../../actions/authActions';
import Google from '../auth/Google';
import Facebook from '../auth/Facebook'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      emailError: '',
      passwordError: '',
      notFoundUser: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      if (nextProps.errors.email) {
        this.setState({ emailError: nextProps.errors.email });
      } else if (nextProps.errors.password) {
        this.setState({ passwordError: nextProps.errors.password });
      } else {
        this.setState({ notFoundUser: nextProps.errors.error });
      }
    } else {
      window.localStorage.setItem('token', loginUser.nextProps.user.token);
    }
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit(e) {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userData, this.props.history);
  }

  render() {
    const { emailError, passwordError, notFoundUser } = this.state;
    return (
      <section id="login" className="flex-grow-1">
        <div className="container">
          <div className="row">
            <div className="col-md-8 m-auto">
              <h3 className="display-4 text-center">Welcome</h3>
              <p className="lead text-center">
                Sign in to access your personalized homepage, follow authors and
                topics you love, and clap for stories that matter to you.
              </p>
              <form noValidate onSubmit={this.onSubmit}>
                <span>{notFoundUser}</span>
                <div className="form-group">
                  <input
                    type="email"
                    className="form-control form-control-lg"
                    placeholder="Email Address"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChange}
                  />
                  {emailError && (
                    <div className="invalid-feedback">{emailError}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    className="form-control form-control-lg"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                  />
                  {passwordError && (
                    <div className="invalid-feedback">{passwordError}</div>
                  )}
                </div>
                <input
                  type="submit"
                  className="btn btn-primary btn-block mt-4 zeus-color"
                />
              </form>
              <div className="mt-2 text-right">
                <p className="lead">
                  <Link to="/passwordreset">Forgot Your Password?</Link>
                </p>
              </div>
              <div className="mt-4 text-center">
                <p className="lead">
                  Or, sign in using your favorite social media account
                </p>
                <Link to="#" className="btn btn-outline-primary mr-1">
                  <Google />
                </Link>
                <Link to="#" className="btn btn-outline-primary mr-1">
                  <Facebook />
                </Link>
              </div>
              <div className="mt-4 text-right">
                <p className="lead">
                  Don't have an account? <Link to="/register">Sign up</Link>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
Login.propType = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};
const mapStateToProps = state => {
  return {
    auth: state.auth,
    errors: state.errors
  };
};

export default connect(
  mapStateToProps,
  { loginUser }
)(withRouter(Login));
