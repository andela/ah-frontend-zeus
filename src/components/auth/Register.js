import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { registerUser } from '../../actions/authActions';
import { withRouter, Link } from 'react-router-dom';
import Google from '../auth/Google';
import Facebook from '../auth/Facebook';

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      errors: {}
    };
    /**
     * binds this operator to onchange and onSubmit.
     */
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors.errors });
    }
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.name,
      email: this.state.email,
      password: this.state.password
    };
    this.props.registerUser(newUser, this.props.history);
  }

  render() {
    const { errors } = this.state;

    return (
      <section id="register" className="flex-grow-1">
        <div className="container">
          <div className="row">
            <div className="col-md-8 m-auto">
              <h3 className="display-4 text-center">Join Authors Haven</h3>
              <p className="lead text-center">
                Create an account to personalize your homepage, follow your
                favorite authors and publications, applaud stories you love, and
                more.
              </p>
              <form noValidate onSubmit={this.onSubmit}>
                <div className="form-group">
                  <input
                    type="text"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.username
                    })}
                    placeholder="Username"
                    name="name"
                    value={this.state.username}
                    onChange={this.onChange}
                  />
                  {errors.username && (
                    <div className="invalid-feedback">{errors.username}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="email"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.Email
                    })}
                    placeholder="Email Address"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChange}
                  />
                  {errors.Email && (
                    <div className="invalid-feedback">{errors.Email}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.password
                    })}
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                  />
                  {errors.password && (
                    <div className="invalid-feedback">{errors.password}</div>
                  )}
                </div>
                <input
                  type="submit"
                  className="btn btn-primary btn-block mt-4 zeus-color"
                />
              </form>
              <div className="mt-4 text-center">
                <p className="lead">
                  Or, you can sign up using your favorite social media account
                </p>
                <Link to="#" className="btn btn-outline-primary mr-1">
                  <Google/>
                </Link>
                <Link to="#" className="btn btn-outline-primary mr-1">
                  <Facebook/>
                </Link>
              </div>
              <div className="mt-4 text-right">
                <p className="lead">
                  Already have an account? <Link to="/login">Sign in</Link>
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

Register.propType = {
  registerUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { registerUser }
)(withRouter(Register));
