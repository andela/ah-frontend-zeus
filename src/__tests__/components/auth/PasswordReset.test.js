import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { PasswordReset } from '../../../components/auth/PasswordReset';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<PasswordReset/>', () => {
  const passwordResetTester = shallow(
    <PasswordReset passwordReset={jest.fn()} />
  );

  test('should render without crashing', () => {
    expect(passwordResetTester).toMatchSnapshot();
  });

  const props = {
    passwordReset: email => jest.fn(),
    errors: {}
  };

  test('component will recieve a user email as new props', () => {
    const wrapper = shallow(<PasswordReset {...props} />);
    const data = {
      errors: {
        user: {
          email: 'User not found'
        }
      }
    };
    wrapper.instance().componentWillReceiveProps(data);
    // console.log(wrapper.instance().props)
    expect(wrapper.instance().state.errors.user.email).toEqual(
      'User not found'
    );
  });

  test('test component onChange', () => {
    const wrapper = shallow(<PasswordReset {...props} />);
    const instance = wrapper.instance();
    instance.onChange({ target: { name: 'name', value: 'value' } });
    const e = { preventDefault: () => {} };
    instance.onSubmit(e);
  });
});
