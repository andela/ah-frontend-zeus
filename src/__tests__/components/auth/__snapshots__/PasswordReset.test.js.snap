// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordReset/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordReset
    passwordReset={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-8 m-auto"
          >
            <h3
              className="display-4 text-center my-4"
            >
              Request a Password reset link
            </h3>
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Your Email Address"
                  type="email"
                />
              </div>
              <div
                className="invalid-feedback"
              />
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>
        </div>
      </div>,
      "className": "flex-grow-1",
      "id": "passwordreset",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-md-8 m-auto"
          >
            <h3
              className="display-4 text-center my-4"
            >
              Request a Password reset link
            </h3>
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Your Email Address"
                  type="email"
                />
              </div>
              <div
                className="invalid-feedback"
              />
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-md-8 m-auto"
          >
            <h3
              className="display-4 text-center my-4"
            >
              Request a Password reset link
            </h3>
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Your Email Address"
                  type="email"
                />
              </div>
              <div
                className="invalid-feedback"
              />
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="display-4 text-center my-4"
              >
                Request a Password reset link
              </h3>,
              <form
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Your Email Address"
                    type="email"
                  />
                </div>
                <div
                  className="invalid-feedback"
                />
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>,
            ],
            "className": "col-md-8 m-auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Request a Password reset link",
                "className": "display-4 text-center my-4",
              },
              "ref": null,
              "rendered": "Request a Password reset link",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter Your Email Address"
                      type="email"
                    />
                  </div>,
                  <div
                    className="invalid-feedback"
                  />,
                  false,
                  <input
                    className="btn btn-primary btn-block mt-4 zeus-color"
                    type="submit"
                  />,
                ],
                "noValidate": true,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter Your Email Address"
                      type="email"
                    />,
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control form-control-lg",
                      "name": "email",
                      "onChange": [Function],
                      "placeholder": "Enter Your Email Address",
                      "type": "email",
                      "value": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "invalid-feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "btn btn-primary btn-block mt-4 zeus-color",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-8 m-auto"
            >
              <h3
                className="display-4 text-center my-4"
              >
                Request a Password reset link
              </h3>
              <form
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Your Email Address"
                    type="email"
                  />
                </div>
                <div
                  className="invalid-feedback"
                />
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>
          </div>
        </div>,
        "className": "flex-grow-1",
        "id": "passwordreset",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-md-8 m-auto"
            >
              <h3
                className="display-4 text-center my-4"
              >
                Request a Password reset link
              </h3>
              <form
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Your Email Address"
                    type="email"
                  />
                </div>
                <div
                  className="invalid-feedback"
                />
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-md-8 m-auto"
            >
              <h3
                className="display-4 text-center my-4"
              >
                Request a Password reset link
              </h3>
              <form
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Your Email Address"
                    type="email"
                  />
                </div>
                <div
                  className="invalid-feedback"
                />
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="display-4 text-center my-4"
                >
                  Request a Password reset link
                </h3>,
                <form
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter Your Email Address"
                      type="email"
                    />
                  </div>
                  <div
                    className="invalid-feedback"
                  />
                  <input
                    className="btn btn-primary btn-block mt-4 zeus-color"
                    type="submit"
                  />
                </form>,
              ],
              "className": "col-md-8 m-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Request a Password reset link",
                  "className": "display-4 text-center my-4",
                },
                "ref": null,
                "rendered": "Request a Password reset link",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="form-group"
                    >
                      <input
                        className="form-control form-control-lg"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter Your Email Address"
                        type="email"
                      />
                    </div>,
                    <div
                      className="invalid-feedback"
                    />,
                    false,
                    <input
                      className="btn btn-primary btn-block mt-4 zeus-color"
                      type="submit"
                    />,
                  ],
                  "noValidate": true,
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        className="form-control form-control-lg"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter Your Email Address"
                        type="email"
                      />,
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control form-control-lg",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "Enter Your Email Address",
                        "type": "email",
                        "value": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "invalid-feedback",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  false,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "btn btn-primary btn-block mt-4 zeus-color",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
